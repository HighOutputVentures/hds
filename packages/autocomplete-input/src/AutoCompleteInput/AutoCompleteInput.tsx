import { ChevronDownIcon, ChevronUpIcon } from '@chakra-ui/icons';
import { Avatar, BoxProps, HStack, Icon } from '@chakra-ui/react';
import {
  FormContainer,
  FormContainerPartProps,
  FormContainerProps,
} from '@highoutput/hds';
import {
  chakraComponents,
  ChakraStylesConfig,
  GroupBase,
  MultiValue,
  Select,
  SingleValue,
} from 'chakra-react-select';
import React, { useId } from 'react';
import getStyles from './styles';

export interface Item {
  value: string | number;
  label: string;
}

export interface OptionItem extends Item {
  avatar?: string;
}

export interface AutoCompleteInputFieldPartProps
  extends FormContainerPartProps {
  reactChakraStyle?:
    | ChakraStylesConfig<Item, boolean, GroupBase<Item>>
    | undefined;
}

export interface AutoCompleteInputFieldProps
  extends Omit<FormContainerProps, 'partProps'> {
  options: OptionItem[];
  loading?: boolean;
  fieldLabelProps?: Omit<BoxProps, 'children'>;
  required?: boolean;
  placement?: 'auto' | 'top' | 'bottom';
  multiple?: boolean;
  darkMode?: boolean;
  autoFocus?: boolean;
  showDropdownIndicator?: boolean;
  placeholder?: string;
  partProps?: Partial<AutoCompleteInputFieldPartProps>;
  value?: string | string[] | number | number[];
  onChangeValue: (...event: any[]) => void;
}

const AutoCompleteInput = (props: AutoCompleteInputFieldProps) => {
  const {
    value,
    options,
    darkMode,
    placement,
    showDropdownIndicator,
    errorMsg,
    partProps,
    autoFocus,
    disabled,
    onChangeValue,
    multiple,
    loading,
    placeholder,
  } = props;

  const uid = useId();

  const styles = getStyles({
    error: Boolean(errorMsg),
    multiple,
    darkMode: darkMode,
  });

  return (
    <FormContainer {...props}>
      <Select
        options={options}
        chakraStyles={partProps?.reactChakraStyle ?? styles}
        placeholder={placeholder}
        menuPlacement={placement ?? 'auto'}
        isMulti={multiple}
        inputId="auto-complete-input"
        useBasicStyles
        aria-label="auto-complete-input"
        captureMenuScroll
        backspaceRemovesValue
        autoFocus={autoFocus}
        isLoading={loading}
        data-testid={`${uid}-auto-complete-input-field`}
        isSearchable
        isDisabled={disabled}
        isClearable
        components={{
          ClearIndicator: () => null,
          IndicatorSeparator: () => null,
          DropdownIndicator: ({ selectProps }) => {
            const icon = selectProps.menuIsOpen
              ? ChevronUpIcon
              : ChevronDownIcon;
            return showDropdownIndicator ? (
              <Icon as={icon} w={4} h={6} stroke="brand.primary.500" />
            ) : null;
          },
          MultiValueContainer: ({ children, data, ...props }) => (
            <chakraComponents.MultiValueContainer {...props} data={data}>
              <HStack align="center" spacing="5px">
                {data.avatar && (
                  <Avatar width="16px" height="16px" src={data.avatar} />
                )}
                {children}
              </HStack>
            </chakraComponents.MultiValueContainer>
          ),
        }}
        onChange={(options: MultiValue<Item> | (SingleValue<Item> | null)) => {
          return isArray<MultiValue<Item>>(options)
            ? onChangeValue(options.map((o) => o.value))
            : onChangeValue(options?.value ?? null);
        }}
        value={options.filter((option) => {
          return Array.isArray(value)
            ? value.some((val: string | number) => val === option.value)
            : value === option.value;
        })}
      />
    </FormContainer>
  );
};

export default AutoCompleteInput;

function isArray<T>(subject: unknown): subject is T {
  return Array.isArray(subject);
}
