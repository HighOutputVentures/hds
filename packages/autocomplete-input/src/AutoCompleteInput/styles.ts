import { ChakraStylesConfig } from 'chakra-react-select';

export interface Item {
  value: string | number;
  label: string;
}

interface GetStylesProps {
  error?: boolean;
  multiple?: boolean;
  darkMode?: boolean;
  testId?: string;
}

const getStyles = ({
  error,
  darkMode,
  testId,
  multiple,
}: GetStylesProps): ChakraStylesConfig<Item> => {
  return {
    menuList: provided => ({
      ...provided,
      _active: {
        background: 'transparent',
      },
      '&::-webkit-scrollbar': {
        width: '16px',
        scrollBehavior: 'smooth',
      },
      '&::-webkit-scrollbar-thumb': {
        background: '#E2E8F0',
        border: '6px solid rgba(0, 0, 0, 0)',
        backgroundClip: 'padding-box',
        borderRadius: '9999px',
      },
      backgroundColor: darkMode ? '#2E2E2E' : undefined,
      maxHeight: '210px',
    }),
    multiValueLabel: baseStyle => ({
      ...baseStyle,
      color: darkMode ? 'neutrals.100' : 'neutrals.700',
      fontSize: 'paragraphs-xs-default',
      letterSpacing: '0.02em',
    }),

    multiValue: baseStyle => {
      return {
        ...baseStyle,
        backgroundColor: darkMode ? 'neutrals.800' : 'alpha.white.500',
        border: `1px solid`,
        borderColor: error
          ? 'interface.error.700'
          : darkMode
          ? 'neutrals.700'
          : 'neutrals.200',
        borderRadius: '6px',
        padding: '2px 5px',
        height: '24px',
        marginRight: '6px',
        'data-testid': testId,
      };
    },
    valueContainer: provided => ({ ...provided, padding: '0 0' }),
    multiValueRemove: provided => ({
      ...provided,
      _hover: {
        backgroundColor: 'none',
      },
      color: 'neutrals.400',
    }),
    control: (base, state) => {
      return {
        ...base,
        _focus: {
          borderColor: error ? 'interface.error.700' : 'brand.primary.700',
        },
        _focusVisible: {
          outline: 'none',
        },
        _invalid: {
          borderColor: 'interface.error.700',
          _hover: { borderColor: 'none' },
        },
        _hover: {
          borderColor: 'brand.primary.700',
        },
        border: '1px solid',
        color: 'neutrals.900',
        backgroundColor: darkMode ? 'neutrals.800' : 'alpha.white.500',
        padding: multiple && state.hasValue ? '0px 14px' : '0px 12px',
        height: '44px',
        boxShadow: '0px 1px 2px rgba(16, 24, 40, 0.05)',
        borderRadius: '8px',
      };
    },
    option: (style, { isFocused, isSelected }) => {
      return {
        ...style,
        backgroundColor: isFocused
          ? darkMode
            ? 'neutrals.700'
            : 'gray.50'
          : isSelected
          ? darkMode
            ? 'neutrals.800'
            : 'gray.50'
          : undefined,
        _hover: {
          backgroundColor: darkMode ? '#525252' : 'gray.50',
        },
        _active: {
          backgroundColor: isSelected
            ? darkMode
              ? 'neutrals.700'
              : 'gray.400'
            : undefined,
        },
        color: darkMode ? 'neutrals.100' : 'neutrals.900',
      };
    },
    menu: style => {
      return {
        ...style,
        zIndex: 9,
        backgroundColor: darkMode ? 'neutrals.800' : 'white',
      };
    },
    singleValue: styles => ({
      ...styles,
      color: darkMode ? 'neutrals.100' : 'neutrals.900',
      fontSize: 'paragraph-sm-default',
    }),
  };
};

export default getStyles;
