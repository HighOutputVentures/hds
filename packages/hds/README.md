# HDS

We aim to build a library of custom ReactJS components that implements our unique UI design conventions (HDS). Our components will be based primarily on Chakra UI. React Storybook will be used for documentation and testing.

- Reusable block of components
- HDS integrated styling
- Easy to implement

## Installation

```bash
npm i @highoutput/hds
```

In your main app file import theme provider and make it a parent of all components.

```tsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ThemeProvider } from '@highoutput/hds';

const App = () => {
  return (<ThemeProvider>{...components here}</ThemeProvider>);
};

ReactDOM.render(<App />, document.getElementById('root'));
```

After that you can use all the components within hds core.

```jsx
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { InputField, Box } from '@highoutput/hds';

const Form = () => {
  return (
    <Box>
      <InputField
        helperMsg="This is a hint text to help user"
        id="name"
        label="Name"
        onBlur={() => {}}
        onChange={() => {}}
        onPressEnter={() => {}}
        placeholder="Input your name"
      />
    </Box>
  );
};

export default Form;
```

## Check out for more information

- [Storybook Documentation and Preview](https://hds-highoutput.vercel.app/?path=/story/getting-started--page)
- [Chakra Components](https://chakra-ui.com/docs/components)

## Other Packages

- [@highoutput/hds-icons](https://www.npmjs.com/package/@highoutput/hds-icons)
- [@highoutput/hds-auth](https://www.npmjs.com/package/@highoutput/hds-auth)
- [@highoutput/hds-spinner](https://www.npmjs.com/package/@highoutput/hds-spinner)
- [@highoutput/hds-pagination](https://www.npmjs.com/package/@highoutput/hds-pagination)
- [@highoutput/hds-array-field](https://www.npmjs.com/package/@highoutput/hds-array-field)
- [@highoutput/hds-auto-form](https://www.npmjs.com/package/@highoutput/hds-auto-form)
- [@highoutput/hds-contact-us](https://www.npmjs.com/package/@highoutput/hds-contact-us)
- [@highoutput/hds-utilities](https://www.npmjs.com/package/@highoutput/hds-utilities)
- [@highoutput/hds-forms](https://www.npmjs.com/package/@highoutput/hds-forms)
- [@highoutput/hds-autocomplete-input](https://www.npmjs.com/package/@highoutput/hds-autocomplete-input)
