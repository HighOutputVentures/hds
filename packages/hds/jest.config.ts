import { Config } from 'jest';

const config: Config = {
  displayName: 'hds',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nrwl/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/hds',
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
};

export default config;
