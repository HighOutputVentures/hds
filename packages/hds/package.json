{
  "version": "0.1.14",
  "license": "MIT",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "start": "tsdx watch",
    "build": "tsdx build",
    "test": "tsdx test --passWithNoTests",
    "lint": "tsdx lint",
    "prepare": "tsdx build",
    "size": "size-limit",
    "analyze": "size-limit --why",
    "generate:types": "chakra-cli tokens ./src/theme/index.ts"
  },
  "peerDependencies": {
    "react": ">=16"
  },
  "husky": {
    "hooks": {
      "pre-commit": "tsdx lint"
    }
  },
  "prettier": {
    "printWidth": 80,
    "semi": true,
    "singleQuote": true,
    "trailingComma": "es5"
  },
  "name": "@highoutput/hds",
  "author": {
    "name": "Niel Berongoy"
  },
  "module": "dist/hds.esm.js",
  "size-limit": [
    {
      "path": "dist/hds.cjs.production.min.js",
      "limit": "10 KB"
    },
    {
      "path": "dist/hds.esm.js",
      "limit": "10 KB"
    }
  ],
  "devDependencies": {
    "@size-limit/preset-small-lib": "7.0.8",
    "size-limit": "7.0.8"
  },
  "jest": {
    "moduleNameMapper": {
      "\\.(css|less|sass|scss)$": "<rootDir>/__mocks__/styleMock.js",
      "\\.(gif|ttf|eot|svg)$": "<rootDir>/__mocks__/fileMock.js"
    }
  },
  "gitHead": "220c468183d0032ba2f8eb5b26d89cf5712292c0",
  "dependencies": {
    "@chakra-ui/icons": "^2.0.2",
    "@chakra-ui/react": "^2.2.1",
    "@highoutput/hds-icons": "^0.1.5",
    "framer-motion": "^7.6.4",
    "lodash": "^4.17.21",
    "lodash-es": "^4.17.21"
  }
}
