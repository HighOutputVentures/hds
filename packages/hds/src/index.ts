export {
  AbsoluteCenter,
  Accordion,
  AccordionButton,
  AccordionIcon,
  AccordionItem,
  AccordionPanel,
  AccordionProvider,
  Alert,
  AlertDescription,
  AlertDialog,
  AlertDialogBody,
  AlertDialogCloseButton,
  AlertDialogContent,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogOverlay,
  AlertIcon,
  AlertTitle,
  AspectRatio,
  Avatar,
  AvatarBadge,
  AvatarGroup,
  Badge,
  Box,
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  Button,
  ButtonGroup,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  Center,
  ChakraBaseProvider,
  ChakraProvider,
  Checkbox,
  CheckboxGroup,
  CheckboxIcon,
  Circle,
  CircularProgress,
  CircularProgressLabel,
  CloseButton,
  Code,
  Collapse,
  ColorModeScript,
  Container,
  Divider,
  Drawer,
  DrawerBody,
  DrawerCloseButton,
  DrawerContent,
  DrawerFooter,
  DrawerHeader,
  DrawerOverlay,
  Editable,
  EditableInput,
  EditablePreview,
  EditableTextarea,
  Fade,
  Flex,
  FormControl,
  FormControlOptions,
  FormErrorIcon,
  FormErrorMessage,
  FormHelperText,
  FormLabel,
  GenericAvatarIcon,
  Grid,
  GridItem,
  HStack,
  Heading,
  Hide,
  Highlight,
  Icon,
  IconButton,
  Image,
  Input,
  InputAddon,
  InputGroup,
  InputLeftAddon,
  InputLeftElement,
  InputRightAddon,
  InputRightElement,
  Kbd,
  LightMode,
  Link,
  LinkBox,
  LinkOverlay,
  List,
  ListIcon,
  ListItem,
  Menu,
  MenuButton,
  MenuCommand,
  MenuDivider,
  MenuGroup,
  MenuIcon,
  MenuItem,
  MenuItemOption,
  MenuList,
  MenuOptionGroup,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  NumberDecrementStepper,
  NumberIncrementStepper,
  NumberInput,
  NumberInputField,
  NumberInputStepper,
  OrderedList,
  PinInput,
  PinInputField,
  Popover,
  PopoverAnchor,
  PopoverArrow,
  PopoverBody,
  PopoverCloseButton,
  PopoverContent,
  PopoverFooter,
  PopoverHeader,
  PopoverTrigger,
  Portal,
  PortalManager,
  Progress,
  ProgressLabel,
  Radio,
  RadioGroup,
  RangeSlider,
  RangeSliderFilledTrack,
  RangeSliderMark,
  RangeSliderProvider,
  RangeSliderThumb,
  RangeSliderTrack,
  ScaleFade,
  Select,
  Show,
  SimpleGrid,
  Skeleton,
  SkeletonCircle,
  SkeletonText,
  Slide,
  SlideFade,
  Slider,
  SliderFilledTrack,
  SliderMark,
  SliderThumb,
  SliderTrack,
  Spacer,
  Spinner,
  Square,
  Stack,
  StackDivider,
  StackItem,
  Stat,
  StatArrow,
  StatDownArrow,
  StatGroup,
  StatHelpText,
  StatLabel,
  StatNumber,
  StatUpArrow,
  Step,
  StepDescription,
  StepIcon,
  StepIndicator,
  StepNumber,
  StepSeparator,
  StepStatus,
  StepStatusType,
  StepTitle,
  Stepper,
  Switch,
  Tab,
  TabIndicator,
  TabList,
  TabPanel,
  TabPanels,
  Table,
  TableCaption,
  TableContainer,
  Tag,
  TagCloseButton,
  TagLabel,
  TagLeftIcon,
  TagRightIcon,
  Tbody,
  Td,
  Text,
  Textarea,
  Tfoot,
  Th,
  Thead,
  Tooltip,
  Tr,
  UnorderedList,
  VStack,
  VisuallyHidden,
  Wrap,
  WrapItem,
  chakra,
  createStylesContext,
  createToastFn,
  cssVar,
  extendBaseTheme,
  extendTheme,
  forwardRef,
  keyframes,
  omitThemingProps,
  shouldForwardProp,
  useBoolean,
  useBreakpoint,
  useBreakpointValue,
  useClipboard,
  useColorModePreference,
  useColorModeValue,
  useConst,
  useControllableProp,
  useControllableState,
  useDimensions,
  useDisclosure,
  useFormControl,
  useMediaQuery,
  useMergeRefs,
  useMultiStyleConfig,
  useOutsideClick,
  usePrefersReducedMotion,
  useStyleConfig,
  useTheme,
  useToast,
  type AbsoluteCenterProps,
  type AccordionButtonProps,
  type AccordionIconProps,
  type AccordionItemProps,
  type AccordionPanelProps,
  type AccordionProps,
  type AlertDescriptionProps,
  type AlertDialogProps,
  type AlertIconProps,
  type AlertProps,
  type AlertStatus,
  type AlertTitleProps,
  type As,
  type AspectRatioProps,
  type AvatarBadgeProps,
  type AvatarGroupProps,
  type AvatarProps,
  type BadgeProps,
  type BoxProps,
  type BreadcrumbItemProps,
  type BreadcrumbLinkProps,
  type BreadcrumbProps,
  type BreadcrumbSeparatorProps,
  type ButtonGroupProps,
  type ButtonProps,
  type CardBodyProps,
  type CardFooterProps,
  type CardHeaderProps,
  type CardProps,
  type CenterProps,
  type ChakraProviderProps,
  type CheckboxGroupProps,
  type CheckboxIconProps,
  type CheckboxProps,
  type CircularProgressLabelProps,
  type CircularProgressProps,
  type CloseButtonProps,
  type CodeProps,
  type CollapseProps,
  type ColorModeScriptProps,
  type ContainerProps,
  type DividerProps,
  type DrawerContentProps,
  type DrawerProps,
  type EditableInputProps,
  type EditablePreviewProps,
  type EditableProps,
  type EditableTextareaProps,
  type FadeProps,
  type FlexProps,
  type FormControlProps,
  type FormErrorMessageProps,
  type FormHelperTextProps,
  type FormLabelProps,
  type GridItemProps,
  type GridProps,
  type HTMLChakraProps,
  type HeadingProps,
  type HideProps,
  type HighlightProps,
  type IconButtonProps,
  type IconProps,
  type ImageProps,
  type InputAddonProps,
  type InputElementProps,
  type InputGroupProps,
  type InputLeftAddonProps,
  type InputLeftElementProps,
  type InputProps,
  type InputRightAddonProps,
  type InputRightElementProps,
  type KbdProps,
  type LinkBoxProps,
  type LinkOverlayProps,
  type LinkProps,
  type ListItemProps,
  type ListProps,
  type MenuButtonProps,
  type MenuCommandProps,
  type MenuDividerProps,
  type MenuGroupProps,
  type MenuItemOptionProps,
  type MenuItemProps,
  type MenuListProps,
  type MenuOptionGroupProps,
  type MenuProps,
  type ModalBodyProps,
  type ModalCloseButtonProps,
  type ModalContentProps,
  type ModalFooterProps,
  type ModalHeaderProps,
  type ModalOverlayProps,
  type ModalProps,
  type MultiStyleConfig,
  type NumberDecrementStepperProps,
  type NumberIncrementStepperProps,
  type NumberInputFieldProps,
  type NumberInputProps,
  type NumberInputStepperProps,
  type PinInputFieldProps,
  type PinInputProps,
  type PopoverArrowProps,
  type PopoverBodyProps,
  type PopoverCloseButtonProps,
  type PopoverContentProps,
  type PopoverFooterProps,
  type PopoverHeaderProps,
  type PopoverProps,
  type PortalManagerProps,
  type PortalProps,
  type ProgressFilledTrackProps,
  type ProgressLabelProps,
  type ProgressProps,
  type ProgressTrackProps,
  type RadioGroupProps,
  type RadioProps,
  type RangeSliderInnerTrackProps,
  type RangeSliderMarkProps,
  type RangeSliderProps,
  type RangeSliderThumbProps,
  type RangeSliderTrackProps,
  type ScaleFadeProps,
  type SelectProps,
  type ShowProps,
  type SimpleGridProps,
  type SkeletonProps,
  type SkeletonTextProps,
  type SlideFadeProps,
  type SlideProps,
  type SliderInnerTrackProps,
  type SliderMarkProps,
  type SliderProps,
  type SliderThumbProps,
  type SliderTrackProps,
  type SpacerProps,
  type SpinnerProps,
  type SquareProps,
  type StackDirection,
  type StackDividerProps,
  type StackProps,
  type StatArrowProps,
  type StatGroupProps,
  type StatHelpTextProps,
  type StatLabelProps,
  type StatNumberProps,
  type StatProps,
  type StepDescriptionProps,
  type StepIndicatorProps,
  type StepProps,
  type StepStatusProps,
  type StepTitleProps,
  type StepperProps,
  type StyleConfig,
  type SwitchProps,
  type TabIndicatorProps,
  type TabListProps,
  type TabPanelProps,
  type TabPanelsProps,
  type TabProps,
  type TableBodyProps,
  type TableCaptionProps,
  type TableCellProps,
  type TableColumnHeaderProps,
  type TableContainerProps,
  type TableFooterProps,
  type TableHeadProps,
  type TableProps,
  type TableRowProps,
  type TagCloseButtonProps,
  type TagLabelProps,
  type TagProps,
  type TextProps,
  type TextareaProps,
  type ThemingProps,
  type TooltipProps,
  type UseDisclosureProps,
  type UseDisclosureReturn,
  type UseFormControlProps,
  type WrapItemProps,
  type WrapProps,
} from '@chakra-ui/react';
export { Global, type GlobalProps } from '@emotion/react';
export { motion } from 'framer-motion';
export { default as ThemeProvider, type ThemeProviderProps } from './ThemeProvider';
export { default as theme } from './theme';
