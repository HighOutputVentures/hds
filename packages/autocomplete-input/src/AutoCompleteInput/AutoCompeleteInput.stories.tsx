import { Flex } from '@chakra-ui/react';
import { ThemeProvider } from '@highoutput/hds';
import { useArgs } from '@storybook/addons';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import AutoCompleteInput from './AutoCompleteInput';

export default {
  title: 'Components/Form Fields/Auto Complete Input',
  component: AutoCompleteInput,
} as ComponentMeta<typeof AutoCompleteInput>;

const Template: ComponentStory<typeof AutoCompleteInput> = (props) => {
  const [args, setArgs] = useArgs();

  const onChange = (value: any[]) => {
    setArgs({ ...args, value });
  };

  return (
    <ThemeProvider>
      <Flex gap={4} flexDir="column">
        <AutoCompleteInput
          {...props}
          onChangeValue={onChange}
          options={args.options}
          helperMsg="This is a hint to help user"
        />
        <AutoCompleteInput
          {...props}
          onChangeValue={onChange}
          multiple
          value={options.map((item) => item.value)}
          options={args.options}
          helperMsg="This is a hint to help user"
        />
        <AutoCompleteInput
          {...props}
          onChangeValue={onChange}
          multiple
          value={options.map((item) => item.value)}
          options={args.options}
          helperMsg="This is a hint to help user"
        />
        <AutoCompleteInput
          {...props}
          onChangeValue={onChange}
          multiple
          disabled
          value={options.map((item) => item.value)}
          options={args.options}
          helperMsg="This is a hint to help user"
        />
        <AutoCompleteInput
          {...props}
          onChangeValue={onChange}
          multiple
          value={options.map((item) => item.value)}
          options={args.options}
          errorMsg="This is an error message"
        />
      </Flex>
    </ThemeProvider>
  );
};

export const Default = Template.bind({});

const options = [
  {
    value: 'Phoenix',
    label: 'Phoenix',
    avatar:
      'https://assets.teenvogue.com/photos/626abe370979f2c5ace0ab29/16:9/w_2560%2Cc_limit/GettyImages-1352932505.jpg',
  },
  {
    value: 'Olivia',
    label: 'Olivia',
    avatar:
      'https://static01.nyt.com/images/2021/05/21/arts/21review-rodrigo01/merlin_188054001_2a34e77d-e653-488a-a4d9-1a0f1ddf73e4-superJumbo.jpg',
  },
];

Default.args = {
  ...Default.args,
  label: 'Users',
  options: options,
  placeholder: 'Add users',
};
