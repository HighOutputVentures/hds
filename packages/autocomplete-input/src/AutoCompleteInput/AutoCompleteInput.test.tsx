import { ThemeProvider } from '@highoutput/hds';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import AutoCompleteInput from './AutoCompleteInput';

const OPTIONS = ['user1', 'user2', 'user3', 'user4', 'user5'];
const SELECT_OPTION = Object.entries(OPTIONS).map(([value, label]) => ({
  value,
  label,
}));

describe('AutoCompleteInput Component', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <AutoCompleteInput
          label="Users"
          options={SELECT_OPTION}
          onChangeValue={jest.fn()}
        />
      </ThemeProvider>
    );
  });

  it('Should render auto complete input field form container', async () => {
    const formControl = await screen.findAllByTestId(
      ':r1:-form-container-form-control'
    );
    expect(formControl).toHaveLength(1);
  });
  it('Should render auto complete input field label', async () => {
    const formControlLabel = await screen.findAllByTestId(
      ':r4:-form-container-label'
    );
    expect(formControlLabel).toHaveLength(1);
  });
  it('Should render auto complete input field', async () => {
    const autoCompleteInput = await screen.findAllByRole('combobox', {
      name: 'auto-complete-input',
    });

    expect(autoCompleteInput).toHaveLength(1);
  });

  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const component = renderer.create(
        <ThemeProvider>
          <AutoCompleteInput
            label="Users"
            options={SELECT_OPTION}
            onChangeValue={function () {}}
          />
        </ThemeProvider>
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      component.unmount();
    });
  });
});
