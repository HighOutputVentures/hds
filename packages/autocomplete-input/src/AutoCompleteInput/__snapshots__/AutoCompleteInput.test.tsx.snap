// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteInput Component Snapshot Should match snapshot 1`] = `
Array [
  <div
    aria-label="Form Group"
    className="chakra-form-control css-1kxonj9"
    data-testid="hds.form.control"
    role="group"
  >
    <label
      className="chakra-form__label css-1xgte9r"
      data-testid="hds.form.control.label"
      htmlFor="field-:r0:"
      id="field-:r0:-label"
    >
      Users
    </label>
    <div
      className=" css-79elbk"
      onKeyDown={[Function]}
    >
      <span
        className="css-1f43avz-a11yText-A11yText"
        id="react-select-6-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className=" css-fciksa"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-1jen3yf"
        >
          <div
            className=" css-1puyje9"
            id="react-select-6-placeholder"
          >
            Select...
          </div>
          <div
            className=" css-svnjlw"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-6-placeholder"
              aria-expanded={false}
              aria-haspopup={true}
              aria-label="auto-complete-input"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              className=" css-146k1ar-input"
              disabled={false}
              getClassNames={[Function]}
              id="auto-complete-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              role="combobox"
              spellCheck="false"
              tabIndex={0}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className=" css-hfbj6y"
        />
      </div>
    </div>
  </div>,
  <span />,
]
`;
